bluemira/equilibria/optimisation/constraint_funcs.py

```diff
@@ -303,8 +306,9 @@ def __init__(
 
     def f_constraint(self, vector: npt.NDArray[np.float64]) -> npt.NDArray[np.float64]:
         """Constraint function"""
-        n_coils = len(vector)
         currents = self.scale * vector
+
+        n_coils = self.n_CS + self.n_PF
```

Comment:
I think we want to put an error in the init so that we catch the case where n_CS and n_PF are 0. 
As we've mentioned in passing this would error slighly further down thew function without this change anyway if they were 0